src/language-server/language-server.ts

```diff
@@ -0,0 +1,93 @@
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import type { CodeAction, DocumentDiagnosticReport, InitializeResult } from 'vscode-languageserver/node.js'
+import {
+  createClientSocketTransport,
+  createConnection,
+  DocumentDiagnosticReportKind,
+  ProposedFeatures,
+  TextDocuments,
+  TextDocumentSyncKind,
+} from 'vscode-languageserver/node.js'
+
+export const getDebugLspPort = () => {
+  const port = Number(process.env.DEBUG_LSP_PORT)
```

Comment:
Going back a little on my previous suggestion. I think we should make this more specific as it's an env var which someone may have set globally for something else. Maybe `PUYA_TS_DEBUG_LSP_PORT`?